packages/data/src/components/use-select/test/index.js

```diff
@@ -434,6 +434,67 @@ describe( 'useSelect', () => {
 			);
 		} );
 
+		it( 'captures state changes before the subscription is set', () => {
+			registry.registerStore( 'store-1', counterStore );
+			registry.registerStore( 'store-2', counterStore );
+			registry.registerStore( 'store-3', counterStore );
+
+			let childShouldDispatch, setChildShouldDispatch;
+			const selectCount1AndDep = jest.fn();
+
+			class ChildComponent extends Component {
+				componentDidUpdate( prevProps ) {
+					if (
+						this.props.childShouldDispatch &&
+						this.props.childShouldDispatch !==
+							prevProps.childShouldDispatch
+					) {
+						registry.dispatch( 'store-1' ).increment();
+					}
+				}
+
+				render() {
+					return null;
+				}
+			}
+
+			const TestComponent = jest.fn( () => {
```

Comment:
We never look at how many times `TestComponent` was called or what its arguments were, so wrapping it with `jest.fn()` is not needed.